### PR TITLE
alicloud_slb_listener support idle_timeout/request_timeout for http/h…

### PR DESCRIPTION
**Pull request by wzb56**
…ttps listeners, and enable_http2/tls_cipher_policy for https listeners.

TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudSlbListener -timeout=300m -count=1
^@=== RUN   TestAccAlicloudSlbListenersDataSource_http
^@^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_http (34.89s)
=== RUN   TestAccAlicloudSlbListenersDataSource_https
^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_https (15.04s)
=== RUN   TestAccAlicloudSlbListenersDataSource_tcp
^@^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_tcp (30.96s)
=== RUN   TestAccAlicloudSlbListenersDataSource_udp
^@^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_udp (30.22s)
=== RUN   TestAccAlicloudSlbListener_importHttp
^@--- PASS: TestAccAlicloudSlbListener_importHttp (13.41s)
=== RUN   TestAccAlicloudSlbListener_importHttps
^@^@--- PASS: TestAccAlicloudSlbListener_importHttps (13.33s)
=== RUN   TestAccAlicloudSlbListener_importTcp
--- PASS: TestAccAlicloudSlbListener_importTcp (13.29s)
=== RUN   TestAccAlicloudSlbListener_importUdp
^@--- PASS: TestAccAlicloudSlbListener_importUdp (13.34s)
=== RUN   TestAccAlicloudSlbListener_http
^@--- PASS: TestAccAlicloudSlbListener_http (13.64s)
=== RUN   TestAccAlicloudSlbListener_https
^@--- PASS: TestAccAlicloudSlbListener_https (13.23s)
=== RUN   TestAccAlicloudSlbListener_tcp
^@--- PASS: TestAccAlicloudSlbListener_tcp (11.92s)
=== RUN   TestAccAlicloudSlbListener_udp
^@--- PASS: TestAccAlicloudSlbListener_udp (12.15s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	215.498s

### CODE REVIEW

File: alicloud/data_source_alicloud_slb_listeners_test.go [hunk 1]
```diff
@@ -47,6 +97,9 @@ func TestAccAlicloudSlbListenersDataSource_http(t *testing.T) {
 	})
 }
 
+
+
+
 func TestAccAlicloudSlbListenersDataSource_tcp(t *testing.T) {
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_slb_listeners_test.go]: Please remove the redundant blank lines.

File: website/docs/r/slb_listener.html.markdown [hunk 2]
```diff
@@ -104,7 +104,13 @@ The following arguments are supported:
...
-* `established_timeout` - (Optinal) Timeout of tcp established connection idle timeout. Valid value range: [10-900] in seconds. Default to 900.
+* `established_timeout` - (Optinal) Timeout of tcp listener established connection idle timeout. Valid value range: [10-900] in seconds. Default to 900.
+* `idle_timeout` - (Optinal) Timeout of http or https listener established connection idle timeout. Valid value range: [1-60] in seconds. Default to 15.
+* `request_timeout` - (Optinal) Timeout of http or https listener request (which does not get response from backend) timeout. Valid value range: [1-180] in seconds. Default to 60.
+* `enable_http2` - (Optinal) Whether to enable https listener support http2 or not. Valid values are `on` and `off`. Default to `on`.
+* `tls_cipher_policy` - (Optinal)  Https listener TLS cipher policy. Valid values are `tls_cipher_policy_1_0`, `tls_cipher_policy_1_1`, `tls_cipher_policy_1_2`, `tls_cipher_policy_1_2_strict`. Default to `tls_cipher_policy_1_0`. (note: Advantanced feature, only support loadbalancer specification more than slb.s1.small)
+
+
```
Comment (xiaozhu36) [inline website/docs/r/slb_listener.html.markdown]: Please remove the redundant blank lines

**Comment by wzb56** [comment]
TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudSlbListener -timeout=300m -count=1

^@=== RUN   TestAccAlicloudSlbListenersDataSource_http
^@^@^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_http (33.06s)
=== RUN   TestAccAlicloudSlbListenersDataSource_https
^@--- PASS: TestAccAlicloudSlbListenersDataSource_https (12.25s)
=== RUN   TestAccAlicloudSlbListenersDataSource_tcp
^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_tcp (28.61s)
=== RUN   TestAccAlicloudSlbListenersDataSource_udp
^@^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_udp (30.04s)
=== RUN   TestAccAlicloudSlbListener_importHttp
--- PASS: TestAccAlicloudSlbListener_importHttp (14.45s)
=== RUN   TestAccAlicloudSlbListener_importHttps
^@--- PASS: TestAccAlicloudSlbListener_importHttps (12.87s)
=== RUN   TestAccAlicloudSlbListener_importTcp
^@--- PASS: TestAccAlicloudSlbListener_importTcp (11.03s)
=== RUN   TestAccAlicloudSlbListener_importUdp
^@--- PASS: TestAccAlicloudSlbListener_importUdp (9.96s)
=== RUN   TestAccAlicloudSlbListener_http
^@--- PASS: TestAccAlicloudSlbListener_http (10.34s)
=== RUN   TestAccAlicloudSlbListener_https
^@--- PASS: TestAccAlicloudSlbListener_https (14.79s)
=== RUN   TestAccAlicloudSlbListener_tcp
--- PASS: TestAccAlicloudSlbListener_tcp (10.26s)
=== RUN   TestAccAlicloudSlbListener_udp
^@--- PASS: TestAccAlicloudSlbListener_udp (8.86s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	196.593s

**Comment by wzb56** [comment]
TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudSlbListener -timeout=300m -count=1

^@^@=== RUN   TestAccAlicloudSlbListenersDataSource_http
^@^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_http (34.51s)
=== RUN   TestAccAlicloudSlbListenersDataSource_https
^@--- PASS: TestAccAlicloudSlbListenersDataSource_https (14.87s)
=== RUN   TestAccAlicloudSlbListenersDataSource_tcp
^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_tcp (29.85s)
=== RUN   TestAccAlicloudSlbListenersDataSource_udp
^@^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_udp (29.70s)
=== RUN   TestAccAlicloudSlbListener_importHttp
--- PASS: TestAccAlicloudSlbListener_importHttp (14.85s)
=== RUN   TestAccAlicloudSlbListener_importHttps
^@--- PASS: TestAccAlicloudSlbListener_importHttps (12.35s)
=== RUN   TestAccAlicloudSlbListener_importTcp
^@^@--- PASS: TestAccAlicloudSlbListener_importTcp (11.73s)
=== RUN   TestAccAlicloudSlbListener_importUdp
^@--- PASS: TestAccAlicloudSlbListener_importUdp (10.19s)
=== RUN   TestAccAlicloudSlbListener_http
^@--- PASS: TestAccAlicloudSlbListener_http (12.60s)
=== RUN   TestAccAlicloudSlbListener_https
^@--- PASS: TestAccAlicloudSlbListener_https (13.47s)
=== RUN   TestAccAlicloudSlbListener_tcp
^@--- PASS: TestAccAlicloudSlbListener_tcp (11.67s)
=== RUN   TestAccAlicloudSlbListener_udp
^@--- PASS: TestAccAlicloudSlbListener_udp (10.57s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	206.421s

**Comment by wzb56** [comment]
TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudSlbListener -timeout=300m -count=1
^@^@=== RUN   TestAccAlicloudSlbListenersDataSource_http
^@^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_http (34.01s)
=== RUN   TestAccAlicloudSlbListenersDataSource_https
^@--- PASS: TestAccAlicloudSlbListenersDataSource_https (13.75s)
=== RUN   TestAccAlicloudSlbListenersDataSource_tcp
^@^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_tcp (31.24s)
=== RUN   TestAccAlicloudSlbListenersDataSource_udp
^@^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_udp (29.26s)
=== RUN   TestAccAlicloudSlbListener_importHttp
^@--- PASS: TestAccAlicloudSlbListener_importHttp (12.46s)
=== RUN   TestAccAlicloudSlbListener_importHttps
^@--- PASS: TestAccAlicloudSlbListener_importHttps (14.26s)
=== RUN   TestAccAlicloudSlbListener_importTcp
^@--- PASS: TestAccAlicloudSlbListener_importTcp (10.83s)
=== RUN   TestAccAlicloudSlbListener_importUdp
^@--- PASS: TestAccAlicloudSlbListener_importUdp (9.95s)
=== RUN   TestAccAlicloudSlbListener_http
^@^@--- PASS: TestAccAlicloudSlbListener_http (12.94s)
=== RUN   TestAccAlicloudSlbListener_https
^@--- PASS: TestAccAlicloudSlbListener_https (12.24s)
=== RUN   TestAccAlicloudSlbListener_tcp
^@--- PASS: TestAccAlicloudSlbListener_tcp (10.71s)
=== RUN   TestAccAlicloudSlbListener_udp
^@--- PASS: TestAccAlicloudSlbListener_udp (11.81s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	203.500s

**Pull request merged**